ygen/protogen.go

```diff
@@ -447,15 +478,16 @@ func genProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *gen
 
```

Comment:
Above function is getting too long. It could benefit from moving up a level in abstraction. Suggest at least leaving a TODO for this.

Reply:
Done.